packages/transformers/babel/src/config.js

```diff
@@ -1,141 +1,191 @@
+import getFlowOptions from './flow';
+import getTypescriptOptions from './typescript';
+import type {BabelConfig} from './types';
 
 const TYPESCRIPT_EXTNAME_RE = /^\.tsx?/;
-const NODE_MODULES = `${path.sep}node_modules${path.sep}`;
-
-export default async function getBabelConfig(asset: MutableAsset) {
-  // Consider the module source code rather than precompiled if the resolver
-  // used the `source` field, or it is not in node_modules.
-  let pkg = await asset.getPackage();
-  let isSource =
-    !!(
-      pkg &&
-      pkg.source &&
-      (await asset.fs.realpath(asset.filePath)) !== asset.filePath
-    ) || !asset.filePath.includes(NODE_MODULES);
-
-  // Try to resolve a .babelrc file. If one is found, consider the module source code.
-  let babelrc = await getBabelRc(asset, pkg, isSource);
-  isSource = isSource || !!babelrc;
-
-  let result = {};
-  mergeConfigs(result, babelrc);
-
-  // Typescript must use the plugin directly (not the typescript preset) and must
-  // come before preset env, otherwise proposals and nonstandard syntax is not
-  // transformed in time.
-  if (path.extname(asset.filePath).match(TYPESCRIPT_EXTNAME_RE)) {
-    let hasTypescript =
-      babelrc &&
-      (hasPlugin(babelrc.config.presets, [
-        '@babel/typescript',
-        '@babel/preset-typescript'
-      ]) ||
-        hasPlugin(babelrc.config.plugins, [
-          '@babel/transform-typescript',
-          '@babel/plugin-transform-typescript'
-        ]));
-
-    if (!hasTypescript) {
-      mergeConfigs(result, {
-        internal: true,
-        babelVersion: 7,
-        config: {
-          plugins: [
-            [
-              require('@babel/plugin-transform-typescript'),
-              {isTSX: path.extname(asset.filePath) === '.tsx'}
-            ]
-          ]
-        }
+const BABEL_TRANSFORMER_DIR = path.dirname(__dirname);
+
+export async function load(config: IConfig) {
+  let partialConfig = loadPartialConfig({filename: config.searchPath});
+  if (partialConfig && partialConfig.hasFilesystemConfig()) {
+    let {babelrc, config: configjs} = partialConfig;
+    let {canBeRehydrated, dependsOnRelative, dependsOnLocal} = getStats(
+      partialConfig.options
+    );
+
+    // babel.config.js files get required by @babel/core so there's no use in including it for watch mode invalidation
+    if (babelrc != null && configjs == null) {
+      config.setResolvedPath(babelrc);
+    } else if (configjs) {
+      // TODO: warn and invalidate on startup
+    }
+
+    if (babelrc && (await isExtended(/*babelrc*/))) {
+      // TODO: warn and invalidate on startup
+      // TODO: maybe add feature to babel that gives details on extended files?
+    }
+
+    if (dependsOnRelative || dependsOnLocal) {
+      // TODO: warn that you may not see changes to relative or local plugins/presets until their
+      // containing package changes versions. You can run with --no-cache to see changes until ready
+      // to bump version.
```

Comment:
what if we added them as included files that invalidated the config when they changed? I guess the problem is node's `require.cache`?

Reply:
We'd have to resolve all of the files that the resolved file depends on in case those changed to, which seemed like it would add a lot of complexity. Do you think that would be worth it?

Reply:
Oh yeah, also the `require.cache` problem. I'll think about this a little more, but I think the best thing to do for now is warn.